babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -1273,4 +1273,36 @@ private void checkParseInfixCast(String sqlType) {
     final String expected = "SELECT CAST('3700 sec' AS INTERVAL MINUTE)";
     sql(sql).ok(expected);
   }
+
+  @Test void testAlternativeTypeConversionQuery() {
+    final String sql = "select (select foo from bar) (integer) from baz";
+    final String expected = "SELECT CAST((SELECT `FOO`\n"
+        + "FROM `BAR`) AS INTEGER)\n"
+        + "FROM `BAZ`";
+    sql(sql).ok(expected);
+  }
+
+  @Test void testAlternativeTypeConversionQueryFormat() {
+    final String sql = "select (select foo from bar) (time(0) format 'HHhMIm') from baz";
+    final String expected = "SELECT CAST((SELECT `FOO`\n"
+        + "FROM `BAR`) AS TIME(0) FORMAT 'HHhMIm')\n"
+        + "FROM `BAZ`";
+    sql(sql).ok(expected);
+  }
+
+  @Test void testAlternativeTypeConversionQueryInterval() {
+    final String sql = "select (select foo from bar) (interval minute) from baz";
+    final String expected = "SELECT CAST((SELECT `FOO`\n"
+        + "FROM `BAR`) AS INTERVAL MINUTE)\n"
+        + "FROM `BAZ`";
+    sql(sql).ok(expected);
+  }
+
+  @Test void testAlternativeTypeConversionQueryNested() {
+    final String sql = "select ((select foo from bar) (integer)) (char) from baz";
```

Comment:
nit: Can we change this nested test case such that the (integer) cast is after "foo"? That would be a more realistic scenario I think.

Reply:
Sure, updated